src/index.ts

```diff
@@ -112,12 +126,16 @@ async function main() {
   });
 }
 
-async function determineLastBlockHeightToProcess(maximumHeight: number, minimumConfirmations: number): Promise<number> {
+async function determineLastBlockHeightToProcess (maximumHeight: number, minimumConfirmations: number, blockDataSource: string): Promise<number> {
   // Finality is understood to be max height minus k blocks. unsafe to process blocks above maxHeight since they could change if there is a long running, short-range fork
   // Alternatively, stop processing at maximum height if lower than finality
   // TODO: where does this really belong?
   let maximum = 0
-  const finalityHeight = await getLatestHeight() - minimumConfirmations;
+  // TODO #13 get "getBlocks" and "getLatestHeight" based on data souce
+  const finalityHeight = (blockDataSource == "MINAEXPLORER") ?
+    await getLatestHeightFromMinaExplorer() - minimumConfirmations : 
```

Comment:
This is not working. Test run shows maximum height is not a number... 
"This script will payout from block 0 to maximum height NaN"

Reply:
fixed - return is an array of 1 block, not just a height.